Anlab.Mvc/Views/Admin/MailQueue.cshtml

```diff
@@ -0,0 +1,125 @@
+﻿@model List<Anlab.Core.Domain.MailMessage>
+@{
+    ViewData["Title"] = "Mail Queue";
+}
+<div id="collapsebutton" class="nodisp expandcollapse btn btn-small btn-success no-print" style="margin: 10px 0px;"><i class="glyphicon glyphicon-minus"></i> Collapse All</div>
+<div id="expandbutton" class="disp expandcollapse btn btn-small btn-success no-print" style="margin: 10px 0px;"><i class="glyphicon glyphicon-plus"></i> Expand All</div>
+<div  style="padding-bottom:120px;">
+<table id="table" cellpadding="0" class="row-border stripe order-column">
+    <thead>
+        <tr>
+            <th>Sent:</th>
+            <th>To:</th>
+            <th>Subject:</th>
+            <th>Sent At:</th>
+
+        </tr>
+    </thead>
+    <tbody style="vertical-align: top;">
+    @foreach (var message in Model)
+    {
+        <tr>
+            <td>
+
+                @if(message.Sent.HasValue && message.Sent == true)
+                {
+                    <p>Yes</p>
+                }
+                else
+                {
+                    <p>No</p>
+                }
+            </td>
+            <td>@message.SendTo
+                
+            </td>
+            <td data-search="@message.Subject">
+                @message.Subject<br />
+            <a data-toggle="collapse" href="#@message.Id" aria-expanded="true" aria-controls="@message.Id">
+                Expand Body
+            </a> |
+            <a href="ViewMessage/@message.Id" target="_blank">Open in new window</a>
```

Comment:
@laholstege To make sure routing works right in all cases you should use a tag helper here, something like this `<a asp-   asp-action="ViewMessage" asp-route-id="@message.Id"> ... </a>`